### PR TITLE
Add densmap embed method and fix binning-summary cluster column bug

### PR DESCRIPTION
**Pull request by WiscEvan**
- :snake: Add densmap `--embed-method` to `autometa-kmers` entrypoint
- :snake: Add trimap `--embed-method` to `autometa-kmers` entrypoint
- :memo: Add `TODO` comments for easy addition of denSNE when it is easily available through conda or pip installation 
- :bug: Change hardcoded `'cluster'` column in `autometa-binning-summary` to `cluster_col` variable

Some comparisons of these methods are explored here: https://github.com/WiscEvan/embed_methods

### CODE REVIEW

**Comment by WiscEvan** [comment]
Some comparisons of these methods are explored here: https://github.com/WiscEvan/embed_methods

**Comment by Sidduppal** [comment]
Trimap runs successfully however, I'm getting an error on running densmap.
Command run:
`autometa-kmers --fasta 78mbp_metagenome.bacteria.fna --kmers 78mbp_metagenome.kmers.tsv --size 5 --norm-output 78mbp_metagenome.kmers.normalized.tsv --embedding-output 78mbp_metagenome.kmers.embedded_densmap.tsv --embedding-method densmap`
Error:
`Traceback (most recent call last):
  File "/home/sidd/miniconda3/envs/autometa/bin/autometa-kmers", line 33, in <module>
    sys.exit(load_entry_point('Autometa==2.0a0', 'console_scripts', 'autometa-kmers')())
  File "/home/sidd/miniconda3/envs/autometa/lib/python3.8/site-packages/Autometa-2.0a0-py3.8.egg/autometa/common/kmers.py", line 738, in main
    embedded_df = embed(
  File "/home/sidd/miniconda3/envs/autometa/lib/python3.8/site-packages/Autometa-2.0a0-py3.8.egg/autometa/common/kmers.py", line 609, in embed
    X = dispatcher[method](**method_args)
  File "/home/sidd/miniconda3/envs/autometa/lib/python3.8/site-packages/Autometa-2.0a0-py3.8.egg/autometa/common/kmers.py", line 588, in do_UMAP
    return UMAP(
TypeError: __init__() got an unexpected keyword argument 'densmap'`

File: autometa/common/kmers.py [hunk 1]
```diff
@@ -42,6 +42,10 @@
 from sklearn.manifold import TSNE
 from tsne import bh_sne
 from umap import UMAP
+import trimap
+
```
Comment (Sidduppal) [inline autometa/common/kmers.py:46]: Shouldn't densmap be imported as well?

File: docs/source/install.rst [hunk 2]
```diff
@@ -31,6 +31,8 @@ Download and install miniconda_. Now run the following commands:
     make create_environment
     # activate autometa conda environment
     conda activate autometa
+    # install trimap for kmer embedding
+    pip install trimap
```
Comment (Sidduppal) [inline docs/source/install.rst:35]: `pip install densmap-learn` should also be added.

File: docs/source/install.rst [hunk 3]
```diff
@@ -49,13 +51,17 @@ Download and install miniconda_. Now run the following commands:
     cd Autometa
     # Construct the autometa environment from requirements.txt
     conda create -n autometa -c conda-forge -c bioconda --file=requirements.txt
+    # Activate environment
+    conda activate autometa
+    # install trimap for kmer embedding
+    pip install trimap
```
Comment (Sidduppal) [inline docs/source/install.rst:57]: `pip install densmap-learn` should also be added.

**Review by Sidduppal — CHANGES_REQUESTED** [review]

**Comment by Sidduppal** [comment]
Step by step tutorial under documentation was not changed ([link](https://github.com/KwanLab/Autometa/blob/dev/docs/source/step-by-step-tutorial.rst#6-k-mer-counting)). You should mention the new options available in the usage table and maybe link to your Jupyter notebook (under advanced usage section) so the user can visually see the difference between them as well?

File: docs/source/install.rst (hunk 3 above)
Comment (WiscEvan) [reply docs/source/install.rst:57]: densmap is installed with umap-learn

File: docs/source/install.rst (hunk 2 above)
Comment (WiscEvan) [reply docs/source/install.rst:35]: densmap is installed with umap-learn

File: autometa/common/kmers.py (hunk 1 above)
Comment (WiscEvan) [reply autometa/common/kmers.py:46]: No, this is available through umap

File: docs/source/install.rst (hunk 2 above)
Comment (WiscEvan) [reply docs/source/install.rst:35]: See here for more details: https://umap-learn.readthedocs.io/en/latest/densmap_demo.html#better-preserving-local-density-with-densmap

**Comment by WiscEvan** [comment]
> Step by step tutorial under documentation was not changed ([link](https://github.com/KwanLab/Autometa/blob/dev/docs/source/step-by-step-tutorial.rst#6-k-mer-counting)). You should mention the new options available in the usage table and maybe link to your Jupyter notebook (under advanced usage section) so the user can visually see the difference between them as well?

Whoops. Missed this, thanks 🙏 . Good idea. Will do 👍

**Review by Sidduppal — APPROVED** [review]

**Review by jason-c-kwan — DISMISSED** [review]
Looks good, I just had a few comments regarding comments in the code that could be improved for clarity. Also, can you amend the step-by-step tutorial to include the new options? There may be other parts of the documentation that also need to be amended. I tested the docker build - worked as expected. Also I went through the code installation instructions in a Ubuntu docker image - worked as expected.

File: autometa/common/kmers.py [hunk 4]
```diff
@@ -464,8 +468,10 @@ def embed(
     -----
 
         * `sklearn.manifold.TSNE <https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html#sklearn.manifold.TSNE>`_
-        * `UMAP <https://umap-learn.readthedocs.io/en/latest/>`_
         * `tsne.bh_sne <https://pypi.org/project/tsne/>`_
+        * `UMAP <https://umap-learn.readthedocs.io/en/latest/>`_
+        * `densMAP <https://github.com/hhcho/densvis/tree/master/densmap>`_
```
[the remark this answers is not in the document]
Comment (WiscEvan) [reply autometa/common/kmers.py:473]: The contributor of densMAP to the UMAP library is part of the densvis group, so I assume they added their code.

File: autometa/common/kmers.py [hunk 5]
```diff
@@ -42,6 +42,10 @@
 from sklearn.manifold import TSNE
 from tsne import bh_sne
 from umap import UMAP
+import trimap
+
+# import densne # TODO: this import should be available after installation of denSNE
```
[the remark this answers is not in the document]
Comment (WiscEvan) [reply autometa/common/kmers.py:47]: :fire: got it

**Review by Sidduppal — APPROVED** [review]
Everything looks great 👍 
I ran the embedding step for `trimap` and `densmap` using the docker image, worked as expected!
You can merge it now 😄

**Pull request merged**